The ":freebsd" config_setting is a copy of ":openbsd" and therefore also
matches on OpenBSD, making the "ares_config_h" select() ambiguous there.
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -42,7 +42,7 @@ config_setting(
 
 config_setting(
     name = "freebsd",
-    constraint_values = ["@platforms//os:openbsd"],
+    constraint_values = ["@platforms//os:freebsd"],
 )
 
 copy_file(
